packages/client/src/tlog/verify/merkle.ts

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2023 The Sigstore Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import crypto from 'crypto';
+import { VerificationError } from '../../error';
+import * as sigstore from '../../types/sigstore';
+
+const RFC6962_LEAF_HASH_PREFIX = Buffer.from([0x00]);
+const RFC6962_NODE_HASH_PREFIX = Buffer.from([0x01]);
+
+export function verifyMerkleInclusion(
+  entry: sigstore.TransparencyLogEntry
+): boolean {
+  const inclusionProof = entry.inclusionProof;
+
+  if (!inclusionProof) {
+    throw new VerificationError('tlog entry has no inclusion proof');
+  }
+
+  const logIndex = BigInt(inclusionProof.logIndex);
+  const treeSize = BigInt(inclusionProof.treeSize);
+
+  if (logIndex >= treeSize) {
+    throw new VerificationError('inclusion proof index exceeds size of tree');
+  }
+
+  // Figure out which subset of hashes corresponds to the inner and border
+  // nodes
+  const { inner, border } = decompInclProof(logIndex, treeSize);
+
+  if (inclusionProof.hashes.length != inner + border) {
```

Comment:
should this use string inequality?
```suggestion
  if (inclusionProof.hashes.length !== inner + border) {
```

Reply:
Updated